services/docker-compose.yml

```diff
@@ -221,7 +221,7 @@ services:
       - default
 
   postgres:
-    image: postgres:10.11
+    image: postgres:10.11@sha256:b679ae588cb304011db5548af06d052a96c34867f80af81f1104162ef1f7ec6b
```

Comment:
one question. is that SHA here forever? say this is 10.11.9. Say they overwrite 10.11.9 with a new image (not saying this will happen with postgres, but it already happened with some other images), what happens to this SHA? it will change. is the old one still available forever or does it get pruned out by the dockerhub registry when cleaning out?

Reply:
The idea behind the sha256 is that if they overwrite the 10.11 tag (because tags are mutable), the sha256 layer is still there and you can pull it. I don't think that docker hub deletes these layers.